src/ota/mod.rs

```diff
@@ -104,3 +126,2053 @@ impl Default for DeployStatus {
+        }
+    }
+}
+
+impl Display for OtaStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            OtaStatus::Idle => write!(f, "Idle"),
+            OtaStatus::Init(req) => write!(f, "Init {req}"),
+            OtaStatus::NoPendingOta => write!(f, "NoPendingOta"),
+            OtaStatus::Acknowledged(req) => write!(f, "Acknowledged {req}"),
+            OtaStatus::Downloading(req, progress) => {
+                write!(f, "Downloading {req} progress {progress}")
+            }
+            OtaStatus::Deploying(req, progress) => write!(f, "Deploying {req} {progress}"),
+            OtaStatus::Deployed(req) => write!(f, "Deployed {req}"),
+            OtaStatus::Rebooting(req) => write!(f, "Rebooting {req}"),
+            OtaStatus::Rebooted => write!(f, "Rebooted"),
+            OtaStatus::Success(req) => write!(f, "Success {req}"),
+            OtaStatus::Error(err, req) => write!(f, "Error {req}: {err}"),
+            OtaStatus::Failure(err, req) => {
+                write!(f, "Failure")?;
+
+                if let Some(req) = req {
+                    write!(f, " {req}")?;
+                }
+
+                write!(f, ": {err}")
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct OtaId {
+    pub uuid: Uuid,
+    pub url: String,
+}
+
+impl From<OtaRequest> for OtaId {
+    fn from(value: OtaRequest) -> Self {
+        Self {
+            uuid: value.uuid.0,
+            url: value.url,
+        }
+    }
+}
+
+impl Display for OtaId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.uuid)
+    }
+}
+
+impl OtaStatus {
+    pub fn ota_request(&self) -> Option<&OtaId> {
+        match self {
+            OtaStatus::Acknowledged(ota_request)
+            | OtaStatus::Downloading(ota_request, _)
+            | OtaStatus::Deploying(ota_request, _)
+            | OtaStatus::Deployed(ota_request)
+            | OtaStatus::Rebooting(ota_request)
+            | OtaStatus::Success(ota_request)
+            | OtaStatus::Error(_, ota_request) => Some(ota_request),
+            OtaStatus::Failure(_, ota_request) => ota_request.as_ref(),
+            _ => None,
+        }
+    }
+}
+
+/// Provides OTA resource accessibility only by talking with it.
+pub struct Ota<T, U>
+where
+    T: SystemUpdate,
+    U: StateRepository<PersistentState>,
+{
+    pub system_update: T,
+    pub state_repository: U,
+    pub download_file_path: PathBuf,
+    pub ota_status: OtaStatus,
+    pub flag: OtaInProgress,
+    pub publisher_tx: mpsc::Sender<OtaStatus>,
+}
+
+#[async_trait]
+impl<T, U> Actor for Ota<T, U>
+where
+    T: SystemUpdate,
+    U: StateRepository<PersistentState>,
+{
+    type Msg = OtaMessage;
+
+    fn task() -> &'static str {
+        "ota"
+    }
+
+    async fn init(&mut self) -> stable_eyre::Result<()> {
+        self.ota_status = OtaStatus::Rebooted;
```

Comment:
The first status should always be `Idle`. The `Rebooted` status indicates that an OTA is in progress and the system has entered the Rebooted status. If you set the `Rebooted` status as the default, it implies that after a reboot, there is always an OTA operation pending. However, a reboot can originate from different system behaviors.

Reply:
If you see here in master:

https://github.com/edgehog-device-manager/edgehog-device-runtime/blob/aff8ebd19811354c1cc8d4ce1c8d494f8eafac91/src/lib.rs#L103

During the creation of the DeviceRuntime struct, we ensure that the pending ota is finished. 

This meeasn sending an EnsurePendingOta message to the ota task: 

https://github.com/edgehog-device-manager/edgehog-device-runtime/blob/aff8ebd19811354c1cc8d4ce1c8d494f8eafac91/src/ota/ota_handle.rs#L158-L164

Which starts the ota event loop with the status Rebooted.

So the initial status should actually be Rebooted since we have to ensure we didn't reboot for an OTA update. After we have checked the state repository, we return a NoPending OTA (like what master does with the ensure_no_pending_ota) and return to the Idle state.